### PR TITLE
ZoneType and ZoneStatus enums

### PR DESCRIPTION
**Pull request by dulitz**
This changes the loose ZONE_STATUS_* constants to the ZoneStatus enum and the loose ZONE_TYPE_* constants to the ZoneType enum.

I renamed a bunch of the zone types to be more generalized. I kept the model-specific names where they seemed to be needed. If you think we should have kept more model-specific names, we can add them as aliases to the ZoneType enum in a subsequent commit.

There are also two random fixes. One is to reauthenticate if we get ResultCode -30002 (invalid session ID). The other is to make
```
   from total_connect_client import TotalConnectClient, ArmType
```
actually work like it's supposed to.

tox passes.

### CODE REVIEW

**Comment by dulitz** [comment]
Oh and one more random fix. One of the police zones but not the other had been included in is_button(). Now they both are.

File: tests/test_client_misc.py [hunk 1]
```diff
@@ -51,7 +52,7 @@ def tests_zone_status(self):
             )
 
             # ask for status of zone 1, which exists
-            assert self.client.zone_status(self.location_id, "1") is ZONE_STATUS_NORMAL
+            assert self.client.zone_status(self.location_id, "1") == ZoneStatus.NORMAL
```
Comment (dulitz) [inline tests/test_client_misc.py:55]: "is" compares identity, which isn't really what you want for numbers (1 is not 1.0 but 1 == 1.0 as you'd expect). Whether enum members are singleton objects (so equality implies identity) or not (so equality doesn't imply identity) seems like an implementation detail. Either way what we are really after here is equality so we use ==.

File: tests/test_zone.py [hunk 2]
```diff
@@ -247,10 +237,10 @@ def tests_type_button(self):
         self.assertTrue(self.zone_button.is_type_button())
         self.assertFalse(self.zone_smoke.is_type_button())
         self.assertFalse(self.zone_gas.is_type_button())
+        self.assertTrue(self.zone_lyric_police.is_type_button())
```
Comment (dulitz) [inline tests/test_zone.py:240]: There is no guarantee that the silent zone type or the audible zone type is or isn't a button, just as there's no guarantee that the monitor zone type is not a button. (And Lyric "local alarm" could be anything as far as I know.)

But more often than not the police zone will be a button.

File: total_connect_client/zone.py [hunk 3]
```diff
@@ -128,37 +152,44 @@ def is_type_button(self):
...
 
     def is_type_motion(self):
         """Return true if zone type is motion."""
-        return self.zone_type_id == ZONE_TYPE_LYRIC_MOTION
+        return self.zone_type_id == ZoneType.INTERIOR_FOLLOWER
 
     def is_type_fire(self):
         """Return true if zone type is fire or smoke."""
-        return self.zone_type_id in (ZONE_TYPE_FIRE_SMOKE, ZONE_TYPE_LYRIC_TEMP)
+        # TODO: why is ZoneType.MONITOR here? it's not a fire or smoke zone type
```
Comment (austinmroczek) [inline total_connect_client/zone.py:186]: I think there are two different types of "temperature" sensors.  One type is to monitor for a low or high temperature point (e.g. freezing) and report back if that point is passed.  Another type is for a very rapid temperature rise, which is meant to indicate fire.

**Comment by austinmroczek** [comment]
fixes #137

File: total_connect_client/zone.py (hunk 3 above)
Comment (dulitz) [reply total_connect_client/zone.py:186]: Right. That second type should be on the fire/smoke zone (which is the only zone not clearable from the keypad I think).

**Pull request merged**